src/Compiler/Compiler.cpp

```diff
@@ -321,10 +319,10 @@ namespace Ark
         }
 
         // put inst and number of arguments
-        page(p).emplace_back(specific);
-        if (specific == Instruction::LIST)
+        page(p).emplace_back(inst);
+        if (inst == Instruction::LIST)
             pushNumber(argc, &page(p));
-        else
+        else if (inst == Instruction::APPEND || inst == Instruction::APPEND_IN_PLACE || inst == Instruction::CONCAT || inst == Instruction::CONCAT_IN_PLACE)
```

Comment:
This might also be the work of another bigger PR (only for the compiler?) because if we change this here, we might also do the same to other places but this isn't the scope of the PR

Reply:
Yes, I have a tendency to digress.